### PR TITLE
Updated read_dicom_file and added unit tests.

### PR DESCRIPTION
**Pull request by BurdManMan**
I fixed a bug in read_dicom_file where it would allow an invalid file to be read. Also added a check if the file being read is empty.

I also created unit tests for read_dicom_file.py.

## Summary by Sourcery

Improve DICOM file reading robustness by adding file validation checks and comprehensive error handling

Bug Fixes:
- Added check to prevent reading empty files
- Added validation for DICOM files with missing required fields

Enhancements:
- Improved error logging for DICOM file reading
- Enhanced file reading to handle edge cases

Tests:
- Created comprehensive unit tests for read_dicom_file function
- Added test cases for valid, headerless, invalid, and empty DICOM files

### CODE REVIEW

**Review by Kahreiru — APPROVED** [review]
don't see any issues with this

File: src/read_dicom_file.py [hunk 1]
```diff
@@ -4,38 +4,48 @@
...
         return dicom_file
 
     except FileNotFoundError as e:
         logger.error("File not found: %s - %s", file_path, str(e))
         return None
 
     except InvalidDicomError as e:
-        logger.warning("File %s is not a valid DICOM file: %s",
-                       file_path, str(e))
+        logger.warning("File %s is not a valid DICOM file: %s", file_path, str(e))
```
Comment (sjswerdloff) [inline src/read_dicom_file.py]: Given that you are logging that it's not a valid DICOM file, but you are going to try to read it with force=True, you might want to clarify that it's not a valid DICOM Part 10 file (at this point).

**Review by sjswerdloff — APPROVED** [review]
very nice test module
